pkg/client/fieldvalidation.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
```

Comment:
Doesn't have to block this PR, but I think we should find a more scalable pattern to wrap clients.

If we continue to introduce a new wrapper for every single option we will end up with a lot of wrappers.

I guess the same way we could either:
* only have one wrapper taking various options and acting accordingly
  * This also makes the options more discoverable for users
  * Pretty sure we end up with more consistency then (see the list of wrappers below)
* add options to the main client (that's probalby not good because then they cannot be used with the fake client)


As of today we have at least:
* intercept.NewClient
* NewDryRunClient
* WithFieldOwner
* WithStrictFieldValidation

Reply:
I completely agree; it's time to find a more scalable pattern to wrap clients.

It would be a breaking change, and will require some back-and-forth. I'll open an issue.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/issues/2888

Reply:
Thx!